### PR TITLE
(performance)[scanner] Isolate local and remote queries using different scanner…

### PR DESCRIPTION
**Pull request by luwei16**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:
Scanner thread pool shared by remote data and local data query. The high latency of remote data query will occupy the thread pool worker thread and affect the performance of local data query.

Therefore, a remote scanner thread pool for remote data query is added to avoid the impact on local query.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -798,6 +798,11 @@ CONF_Int32(s3_transfer_executor_pool_size, "2");
 
 CONF_Bool(enable_time_lut, "true");
 
+// number of s3 scanner thread pool size
+CONF_Int32(doris_s3_scanner_thread_pool_thread_num, "48");
```
Comment (pengxiangyu) [inline be/src/common/config.h:802]: DEFAULT VALUE 48 is too big, it will cost too many cpu

File: be/src/common/config.h [hunk 2]
```diff
@@ -798,6 +798,11 @@ CONF_Int32(s3_transfer_executor_pool_size, "2");
...
+// number of s3 scanner thread pool queue size
+CONF_Int32(doris_s3_scanner_thread_pool_queue_size, "102400");
```
Comment (pengxiangyu) [inline be/src/common/config.h:804]: 102400 is too big,

File: be/src/exec/olap_scan_node.cpp [hunk 3]
```diff
@@ -1503,6 +1503,7 @@ void OlapScanNode::transfer_thread(RuntimeState* state) {
      * 4. Regularly increase the priority of the remaining tasks in the queue to avoid starvation for large queries
      *********************************/
     PriorityThreadPool* thread_pool = state->exec_env()->scan_thread_pool();
+    PriorityThreadPool* s3_thread_pool = state->exec_env()->s3_scan_thread_pool();
```
Comment (pengxiangyu) [inline be/src/exec/olap_scan_node.cpp:1506]: Use remote_thread_pool, it is not only for S3

File: be/src/runtime/exec_env.h [hunk 4]
```diff
@@ -200,6 +201,7 @@ class ExecEnv {
 
     // TODO(cmy): find a better way to unify these 2 pools.
     PriorityThreadPool* _scan_thread_pool = nullptr;
+    PriorityThreadPool* _s3_scan_thread_pool = nullptr;
```
Comment (pengxiangyu) [inline be/src/runtime/exec_env.h:204]: _s3_scan_thread_pool need to be _remote_scan_thread_pool

File: be/src/common/config.h (hunk 1 above)
Comment (luwei16) [reply be/src/common/config.h]: changed to 16

File: be/src/exec/olap_scan_node.cpp (hunk 3 above)
Comment (luwei16) [reply be/src/exec/olap_scan_node.cpp]: done

File: be/src/runtime/exec_env.h (hunk 4 above)
Comment (luwei16) [reply be/src/runtime/exec_env.h]: done

**Review by pengxiangyu — DISMISSED** [review]
LGTM

**Review by kpfly — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Pull request merged**